packages/diagrams/frontend/sirius-components-diagrams/src/renderer/delete/useDiagramDeleteMutation.tsx

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+ * Copyright (c) 2023, 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { gql, useMutation } from '@apollo/client';
+import { useMultiToast } from '@eclipse-sirius/sirius-components-core';
+import { useCallback, useContext, useEffect } from 'react';
+import { DiagramContext } from '../../contexts/DiagramContext';
+import { DiagramContextValue } from '../../contexts/DiagramContext.types';
+import {
+  GQLDeleteFromDiagramData,
+  GQLDeleteFromDiagramInput,
+  GQLDeleteFromDiagramPayload,
+  GQLDeleteFromDiagramSuccessPayload,
+  GQLDeleteFromDiagramVariables,
+  GQLDeletionPolicy,
+  GQLErrorPayload,
+  UseDiagramDeleteMutation,
+} from './useDiagramDeleteMutation.types';
+
+export const deleteFromDiagramMutation = gql`
+  mutation deleteFromDiagram($input: DeleteFromDiagramInput!) {
+    deleteFromDiagram(input: $input) {
+      __typename
+      ... on ErrorPayload {
+        messages {
+          body
+          level
+        }
+      }
+      ... on DeleteFromDiagramSuccessPayload {
+        messages {
+          body
+          level
+        }
+      }
+    }
+  }
+`;
+
+const isErrorPayload = (payload: GQLDeleteFromDiagramPayload): payload is GQLErrorPayload =>
+  payload.__typename === 'ErrorPayload';
+
+const isSuccessPayload = (payload: GQLDeleteFromDiagramPayload): payload is GQLDeleteFromDiagramSuccessPayload =>
+  payload.__typename === 'DeleteFromDiagramSuccessPayload';
+
+export const useDiagramDeleteMutation = (): UseDiagramDeleteMutation => {
```

Comment:
This hook should probably be named `useDeleteFromDiagram` to stay consistent with our naming conventions